src/worker_job.cc

```diff
@@ -160,7 +160,19 @@ namespace detail {
 				const auto [bid, mode] = access_map.get_nth_access(i);
 				const auto sr = grid_box_to_subrange(access_map.get_requirements_for_nth_access(i, tsk->get_dimensions(), data.sr, tsk->get_global_size()));
 				const auto info = m_buffer_mngr.access_host_buffer(bid, mode, sr);
-				access_infos.push_back(closure_hydrator::accessor_info{info.ptr, info.backing_buffer_range, info.backing_buffer_offset, sr});
+
+#if CELERITY_ACCESSOR_BOUNDARY_CHECK
+					// for host tasks this can be a vector (RAII)
+					std::vector<id<3>> oob_idx {2};
```

Comment:
```suggestion
					std::vector<id<3>> oob_indices {2};
```

I think it's clear from context why you can use a vector here instead of doing a `sycl::malloc_shared`. Also, this is not really RAII because you're moving it into `m_oob_indices_per_accessor` a few lines further down.

I missed this in my original review of the device accessor boundary check, but this should really be called `oob_indices` (we use plural form everywhere else). Maybe you could also fix this for the device accessor?